File: C/605-can-place-flowers/main.go
package main

import "fmt"

func canPlaceFlowers(flowerbed []int, n int) bool {
	candi := 0
	for i := 0; i < len(flowerbed); i++ {
		if flowerbed[i] == 1 {
			continue
		}
		previous, later := true, true
		if i > 0 && flowerbed[i-1] == 1 {
			previous = false
		}
		if i < len(flowerbed)-1 && flowerbed[i+1] == 1 {
			later = false
		}
		if previous && later {
			flowerbed[i] = 1
			candi++
		}
	}
	return candi >= n
}

func main() {
	fmt.Println("a")
}

File: C/605-can-place-flowers/main.rs
struct Solution {}

impl Solution {
  pub fn can_place_flowers(flowerbed: Vec<i32>, n: i32) -> bool {
    let mut sum: i32 = 0;
    let mut prev: i32 = -2;
    for i in 0..flowerbed.len() {
      if flowerbed[i] == 1 {
        sum += (i as i32 - prev - 2) / 2;
        prev = i as i32;
      }
    }
    sum += (flowerbed.len() as i32 - prev - 1) / 2;
    sum >= n
  }
}
